File: src/main/java/com/freya02/docs/data/DocDetail.java
package com.freya02.docs.data;

import com.freya02.docs.HTMLElement;
import com.freya02.docs.HTMLElementList;

import java.util.List;

public class DocDetail extends HTMLElementList {
	private final DocDetailType detailType;

	public DocDetail(DocDetailType detailType, List<HTMLElement> htmlElements) {
		super(htmlElements);

		this.detailType = detailType;
	}

	public DocDetailType getDetailType() {
		return detailType;
	}

	public String getDetailString() {
		return detailType.getElementText();
	}
}

File: doxxy-bot/src/main/kotlin/com/freya02/docs/data/DocDetail.kt
package com.freya02.docs.data

import com.freya02.docs.HTMLElement
import com.freya02.docs.HTMLElementList

class DocDetail(detailType: DocDetailType, htmlElements: List<HTMLElement>) : HTMLElementList(htmlElements) {
    val detailString: String = detailType.elementText
}